File: features/advisor/src/main/java/com/nivasafinance/features/advisor/dto/AdvisorResponse.java
package com.nivasafinance.features.advisor.dto;

import com.nivasafinance.features.advisor.entity.Advisor;
import com.nivasafinance.common.enums.ReferredByType;
import com.nivasafinance.features.advisor.enums.AdvisorStatus;
import com.nivasafinance.features.sourcechannel.dto.SourcingChannelResponse;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdvisorResponse {

    private Long id;
    private UUID identifier;
    private AdvisorStatus status;
    private PersonalDetails personalDetails;
    private AdvisorRemarks remarks;
    private QualificationDetails qualificationDetails;
    private OtherDetails otherDetails;
    private SegmentationDetails segmentationDetails;
    private String ownerUsername;
    private String officeKey;
    private String officeName;
    private String referralCode;
    @Deprecated
    private SourcingChannelResponse sourcingChannelDetails;
    private List<Advisor.SourcingEntry> sourcingHistory;

    //referral details
    private String referredByCode;
    private UUID referredByIdentifier;
    private ReferredByType referredByType;
    private String referredByName;
    private String referredByNumber;
    
}

File: features/advisor/src/main/java/com/nivasafinance/features/advisor/dto/AdvisorResponse.kt
package com.nivasafinance.features.advisor.dto

import com.nivasafinance.features.person.dto.PersonResponse
import java.time.LocalDateTime
import java.util.UUID

data class AdvisorResponse(
    val id: UUID,
    val personId: UUID?,
    val person: PersonResponse?,
    val advisorCode: String?,
    val verificationStatus: String?,
    val verificationNotes: String?,
    val extData: Map<String, Any>?,
    val createdAt: LocalDateTime,
    val createdBy: String?,
    val updatedAt: LocalDateTime,
    val updatedBy: String?
)
